### ISSUE TITLE
Simulator: Update interface to support yellow and blue teams

### ISSUE THREAD

**Issue by MathewMacDougall** [UBC-Thunderbots/Software#1326]
### Description of the task

The current interface of the `Simulator` class is for interacting only with teams in terms of `friendly` and `enemy`. This needs to be updated to support arbitrary `yellow` and `blue` teams.

In addition, the `Simulator` will now be less responsible for timing. This will be moved to a new `ThreadedSimulator` or the `SimulatedTestFixture` as needed.
* Simulator will only have `runForDuration` and `getWorld`. Nothing more for timing related code
* For now the output of the simulator will still be out internal datatypes (we will change to proto in a separate issue)

[Design Doc](https://docs.google.com/document/d/1JMWTZS6ekHbWOCAPGTgZ8TorohB0iuXrKP28RS0B6po/edit?usp=sharing)

This will likely require a change to either the SimulatorBackend to handle more timing, or the SimulatedTestFixture.

### Acceptance criteria

- [ ] Update simulator interface
- [ ] Unit tests

### Blocked By

#1325

**Comment by MathewMacDougall** [comment]
For reference, full comment from a brainstorming session
```
/* Current interface is OK
 *
 * No threading in Simulator class. SimulatedTestFixture can take care of managing
 * determinism with other components, don't use threaded observers. Implement the real-time
 * delay in the test fixture if the visualizer is enabled. (visualizer will still need it's own thread).
 * (Possibly worth refactoring Visualizer have threaded and non threaded version).
 *
 * Don't need enums to wait for team primitives, don't need enum for speed mode.
 *
 * Simulator only has runForDuration and getWorld. DOes not care about timing.
 *
 * Any timing related stuff will be iomplemented by ThreadedSimulator and the test fixture (separately)
 *
 * ThreadedSimulator should extend the Simulator so functions don't need to be duplicated for the most part
 *
 * Make separate ThreadedSimulator that runs in real-time
 */

```
